specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-10-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,786 @@
+              "type": "string",
+              "readOnly": true,
+              "$ref": "#/definitions/FluxComplianceStateDefinition"
+            },
+            "provisioningState": {
+              "description": "Status of the creation of the fluxConfiguration.",
+              "type": "string",
+              "readOnly": true,
+              "$ref": "../../common/2021-10-01-preview/definitions.json#/definitions/ProvisioningStateDefinition"
+            },
+            "errorMessage": {
+              "description": "Error message returned to the user in the case of provisioning failure.",
+              "type": "string",
+              "x-nullable": true,
+              "readOnly": true
+            }
+          }
+        },
+        "systemData": {
+          "description": "Top level metadata https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "FluxConfigurationPatch": {
+      "description": "The Flux Configuration Patch Request object.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "Updatable properties of an Flux Configuration Patch Request",
+          "properties": {
+            "sourceKind": {
+              "description": "Source Kind to pull the configuration data from.",
+              "$ref": "#/definitions/SourceKindDefinition"
+            },
+            "suspend": {
+              "description": "Whether this configuration should suspend its reconciliation of its kustomizations and sources.",
+              "type": "boolean",
+              "default": false
+            },
+            "gitRepository": {
+              "description": "Parameters to reconcile to the GitRepository source kind type.",
+              "type": "object",
+              "$ref": "#/definitions/GitRepositoryDefinition"
+            },
+            "kustomizations": {
+              "description": "Array of kustomizations used to reconcile the artifact pulled by the source type on the cluster.",
+              "type": "object",
+              "x-nullable": true,
+              "additionalProperties": {
+                "type": "object",
+                "$ref": "#/definitions/KustomizationDefinition"
+              }
+            },
+            "configurationProtectedSettings": {
+              "description": "Key-value pairs of protected configuration settings for the configuration",
+              "type": "object",
+              "x-nullable": true,
+              "x-ms-secret": true,
+              "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=15&length=20)

additionalProperties pattern is not recommended unless all properties are dynamic or user created. Object should be schematized if possible.